Spec/Auth.swift

```diff
@@ -205,6 +205,272 @@ class Auth : QuickSpec {
+                            options.authParams?.append(NSURLQueryItem(name: "body", value: invalidToken))
+
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Disconnected) { stateChange in
+                                    guard let stateChange = stateChange else {
+                                        fail("ConnectionStateChange is nil"); done(); return
+                                    }
+                                    guard let errorInfo = stateChange.reason else {
+                                        fail("ErrorInfo is nil"); done(); return
+                                    }
+                                    expect(errorInfo.code) == 80019
+                                    done()
+                                }
+                            }
+
+                            guard let errorInfo = realtime.connection.errorReason else {
+                                fail("ErrorInfo is empty"); return
+                            }
+                            expect(errorInfo.code) == 80019
+                        }
+
+                        // RSA4c3
+                        it("if the connection is CONNECTED, then the connection should remain CONNECTED") {
+                            var options = AblyTests.clientOptions()
+                            options.authUrl = NSURL(string: "http://echo.ably.io")!
+                            options.authParams = [NSURLQueryItem]()
+                            options.authParams?.append(NSURLQueryItem(name: "type", value: "text"))
+
+                            let token = getTestToken(ttl: 2.0)
+                            options.authParams?.append(NSURLQueryItem(name: "body", value: token))
+
+                            let realtime = ARTRealtime(options: options)
+                            defer { realtime.dispose(); realtime.close() }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                realtime.connection.once(.Connected) { stateChange in
+                                    expect(stateChange?.reason).to(beNil())
+                                    done()
+                                }
+                            }
+
+                            // Token should renew and fail
+                            let invalidToken = String(token.characters.reverse())
+                            realtime.options.authParams = [NSURLQueryItem]()
+                            realtime.options.authParams?.append(NSURLQueryItem(name: "type", value: "text"))
+                            realtime.options.authParams?.append(NSURLQueryItem(name: "body", value: invalidToken))
+
+                            expect(realtime.connection.errorReason).toEventuallyNot(beNil(), timeout: testTimeout)
```

Comment:
Same here:

> As @mattheworiordan says above, this should happen because of an incoming AUTH and not because the token expires, because the token expiring, as per RTN15a, causes a transition to DISCONNECTED right away instead of staying CONNECTED.